common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java

```diff
@@ -224,6 +224,18 @@ public static void load(String name, ClassLoader loader) {
             out = null;
 
             loadLibrary(loader, tmpFile.getPath(), true);
+        } catch (UnsatisfiedLinkError e) {
+            try {
+                if (tmpFile.isFile() && tmpFile.canRead() && NoexecFileStoreDetector.isOnNoexecFileStore(tmpFile)) {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'tmpFile' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2259)